src/plugins/reference/README.md

```diff
@@ -0,0 +1,128 @@
+- infos = Information about the template plugin is in keys below
+- infos/author = Klemens Böswirth <k.boeswirth+git@gmail.com>
+- infos/licence = BSD
+- infos/needs =
+- infos/provides =
+- infos/recommends =
+- infos/placements = presetstorage
+- infos/status = nodep libc preview experimental unfinished
+- infos/metadata = check/reference check/reference/restrict
+- infos/description = Useful plugin for validating singular or recursive references
+
+## Introduction
+
+This plugin serves one single purpose: the validation of reference values inside of keys.
+
+To specify a key as a reference add the metakey `check/reference`. Currently the metakey 
+supports the three values `single`, `recursive` and `alternative`. While the value 
+`alternative` only makes sense in a certain context inside the `recursive` mode, the other
+two values define to different modes of operation for the plugin. 
+
+If a key has the metakey `check/reference` with value `single`, the plugin reads the value
+of the key and produces an error, if the value is not a valid reference.
+
+If a key has metakey `check/reference` with value `recursive`, the plugin constructs a 
+reference graph starting with the marked key. (for the exact construction see below) 
+In this reference graph each node corresponds to a key in the KDB, while each edge 
+represents a reference between to keys. The plugin will produce an error, if this graph
+is not a directed acyclic graph or contains any invalid references.
+
+Additionally to this basic functionality the plugin allows one to restrict the set of valid
+references using the `check/reference/restrict` metakey. The exact mechanism for this is 
+described below.
+
+### Resolution of references
+If the key `A` is interpreted as a reference by the plugin, it has to be an array. Each of 
```

Comment:
Why does it need to be an array for the key A? Is this an example?

Having everything as array is a bit tailored towards the INI array feature. Aren't references also useful without arrays?

Reply:
You are right, we probably should use the same approach as described for `check/reference/restrict`.